modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -1363,6 +1391,88 @@ public MetadataList updateEventMetadata(String id, MetadataList metadataList, El
+      String publicationId = publication.getIdentifier();
+      URI publicationURI = publication.getURI();
+      MimeType publicationMimeType = publication.getMimeType();
+
+      if (channelId == null || publicationURI == null || publicationMimeType == null || publicationId == null) {
+        // No publication, do nothing
+        continue;
+      }
+
+      // Get metadata and acls
+      SimpleElementSelector elementSelector = new SimpleElementSelector();
+      // This does not necessarily work for extended metadata, which can have completely different flavors.
+//      elementSelector.addFlavor("dublincore/*");
+      // Relies on ACLs having this particular flavor
+      elementSelector.addFlavor("security/*");
+      Collection<MediaPackageElement> elements = elementSelector.select(mediaPackage, false);
+
+      Set<String> elementIds = new HashSet<String>();
+      for (MediaPackageElement elem : elements) {
+        elementIds.add(elem.getIdentifier());
+      }
+
+      // To make sure we hit all extended metadata catalogs, let's just get all catalogs
+      for (Catalog catalog : mediaPackage.getCatalogs()) {
+        elementIds.add(catalog.getIdentifier());
+      }
+
+      if (elementIds.size() < 1) {
+        // Nothing to republish, do nothing
+        continue;
+      }
+
+      // Remove publication from mediapackage
+      for (Publication publicationElement : mediaPackage.getPublications()) {
+        if (channelId.equals(publicationElement.getChannel())) {
+          mediaPackage.remove(publicationElement);
+        }
+      }
+
+      List<MediaPackageElement> downloadElements = new ArrayList<>();
+      List<MediaPackageElement> streamingElements = new ArrayList<>();
+      try {
+        downloadElements = downloadDistributionService.distributeSync(channelId, mediaPackage, elementIds, false);
+        if (streamingDistributionService != null && streamingDistributionService.publishToStreaming()) {
+          streamingElements = streamingDistributionService.distributeSync(channelId, mediaPackage, elementIds);
+        }
+      } catch (DistributionException e) {
+        throw new RuntimeException(e);
+      }
+      logger.debug("Distribute of mediapackage {} to channel {} completed", mediaPackage, channelId);
+
+      // Re-add publication
+      Publication newPublication = PublicationImpl.publication(publicationId, channelId, publicationURI, publicationMimeType);
+
+      // Add published elements
+      for (MediaPackageElement element : downloadElements) {
+        element.setIdentifier(null);
+        PublicationImpl.addElementToPublication(newPublication, element);
+      }
+      for (MediaPackageElement element : streamingElements) {
+        element.setIdentifier(null);
+        PublicationImpl.addElementToPublication(newPublication, element);
+      }
+
+      mediaPackage.add(newPublication);
+    }
+
+    // Also search service
+    try {
+      searchService.addSynchronously(mediaPackage);
```

Comment:
The PublishEngageWOH does a bunch of transformations to the media package before adding it to search, don't we need to do any of those here?

Reply:
Maybe. Seemed superfluous to me.